Engine/Graphics/DecalBuilder.cpp

```diff
@@ -264,60 +263,54 @@ char DecalBuilder::_49B790_build_decal_geometry(
 
     decal->uColorMultiplier = uColorMultiplier;
     decal->uNumVertices = 4;
-    decal->field_C14 = LightLevel;
+    decal->DimmingLevel = LightLevel;
 
-    result = engine->pStru9Instance->_4980B9(
-        a11, a10, FacetNormals->Normal.x, FacetNormals->Normal.y, FacetNormals->Normal.z, decal->pVertices,
-        (signed int *)&decal->uNumVertices);
+    // clip decals to face
+    result = engine->pStru9Instance->ClipVertsToFace(
+        faceverts, numfaceverts, FacetNormals->Normal.x, FacetNormals->Normal.y, FacetNormals->Normal.z, decal->pVertices,
+        (signed int*)&decal->uNumVertices);
 
     if (result) {
         if (!decal->uNumVertices) return 1;
 
-        pIndoorCameraD3D->ViewTransform(decal->pVertices,
-                                        (unsigned int)decal->uNumVertices);
```

Comment:
we dont need this anymore?

Reply:
Viewtransform and project moved to decal drawing.
Dropped the slightly overzealous clipping for now (was needed for sotware rendering).